### ISSUE TITLE
ES6 classes aren't being registered as defined with react/jsx-no-undef

### ISSUE THREAD

**Issue by framerate** [yannickcr/eslint-plugin-react#30]
(First seen in #24)

It would appear that ```react/jsx-no-undef``` isn't working properly (according to @yannickcr that's what I'm seeing here) while using babel-eslint. Here's the problems:

```
class RalphMacchio extends React.Component {
 ...blah blah
}
```

and...

```
React.render(<RalphMacchio />, document.getElementById('rm'));
```

Gives me:

```
src/app.jsx
  41:14  error  'RalphMacchio' is not defined  react/jsx-no-undef
  41:14  error  'RalphMacchio' is not defined  no-undef
```

Forgive me (and point me in the right direction) if this is unrelated. I'm trying to switch to ESLint to support my new ES6 projects. Here's my WIP .eslintrc:

```
{
    "parser": "babel-eslint",
    "env": {
        "node": true,
        "es6": true
    },

    "plugins": [
        "react"
    ],

    "settings": {
        "ecmascript": 6
    },

    "ecmaFeatures": {
        "jsx": true
    },

    "rules": {
        "no-undef": 2,
        "react/jsx-no-undef": 2,
        "no-unused-vars": 2,
        "react/jsx-uses-vars": 1,
        "no-console": 1,
        "eqeqeq": [2, "smart"],
        "quotes": [2, "single"]
    },

    "globals": {
        "document": true,
        "window": true,
        "_": true
    }
}
```

**Comment by framerate** [comment]
Thanks @yannickcr !

**Comment by yannickcr** [comment]
It's just a quick patch, I think the real issue is in `babel-eslint` but I need to investigate.

However, it does not fix the error reported by the ESLint rule `no-undef`, this one should be fixed in next ESLint release eslint/eslint#2122

**Comment by framerate** [comment]
Yeah I saw the notes. This is great news! I've been trying to setup a testbed to work with ESLint (instead of JSHint) and wanted to use ES6 as well... Glad to see the quick progress!

Thanks again!
